exporter/sumologicexporter/exporter.go

```diff
@@ -15,45 +15,296 @@
+	var err error
+
+	if body.Len() > 0 && body.Len()+len(line) > se.config.MaxRequestBodySize {
+		err = se.send(LogsPipeline, body.String(), fields)
+		body.Reset()
+	}
+
+	if body.Len() > 0 {
+		// Do not add newline if the body is empty
+		body.WriteString("\n")
+	}
+
+	body.WriteString(line)
+	return err
+}
+
+func (se *sumologicexporter) sendLogsTextFormat(buffer []pdata.LogRecord, fields string) error {
+	body := strings.Builder{}
+	var errs []error
+
+	for j := 0; j < len(buffer); j++ {
+		err := se.appendAndSend(buffer[j].Body().StringVal(), LogsPipeline, &body, fields)
+		if err != nil {
+			errs = append(errs, err)
+		}
+	}
+
+	err := se.send(LogsPipeline, body.String(), fields)
+	if err != nil {
+		errs = append(errs, err)
+	}
+
+	if len(errs) > 0 {
+		return componenterror.CombineErrors(errs)
+	}
 	return nil
 }
 
-func (se *sumologicexporter) ConsumeLogs(_ context.Context, _ pdata.Logs) error {
+func (se *sumologicexporter) sendLogsJSONFormat(buffer []pdata.LogRecord, fields string) error {
+	body := strings.Builder{}
+	var errs []error
+
+	for j := 0; j < len(buffer); j++ {
+		data := se.filterMetadata(buffer[j].Attributes(), true)
+		data[logKey] = buffer[j].Body().StringVal()
+		nextLine, err := json.Marshal(data)
+
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		err = se.appendAndSend(bytes.NewBuffer(nextLine).String(), LogsPipeline, &body, fields)
+		if err != nil {
+			errs = append(errs, err)
+		}
+	}
+
+	err := se.send(LogsPipeline, body.String(), fields)
+	if err != nil {
+		errs = append(errs, err)
+	}
+
+	if len(errs) > 0 {
+		return componenterror.CombineErrors(errs)
+	}
 	return nil
 }
 
-func (se *sumologicexporter) ConsumeMetrics(_ context.Context, _ pdata.Metrics) error {
+func (se *sumologicexporter) sendLogs(buffer []pdata.LogRecord, fields string) error {
+	if se.config.LogFormat == TextFormat {
+		return se.sendLogsTextFormat(buffer, fields)
+	} else if se.config.LogFormat == JSONFormat {
+		return se.sendLogsJSONFormat(buffer, fields)
+	} else {
+		return errors.New("Unexpected log format")
+	}
+}
+
+// Send sends data to sumologic
+func (se *sumologicexporter) send(pipeline string, body string, fields string) error {
+	// Add headers
+	req, err := http.NewRequest("POST", se.config.URL, strings.NewReader(body))
+
+	if err != nil {
+		return err
+	}
+
+	req.Header.Add("X-Sumo-Client", se.config.Client)
+
+	if len(se.config.SourceHost) > 0 {
+		req.Header.Add("X-Sumo-Host", se.config.SourceHost)
+	}
+
+	if len(se.config.SourceName) > 0 {
+		req.Header.Add("X-Sumo-Name", se.config.SourceName)
+	}
+
+	if len(se.config.SourceCategory) > 0 {
+		req.Header.Add("X-Sumo-Category", se.config.SourceCategory)
+	}
+
+	if pipeline == LogsPipeline {
+		req.Header.Add("Content-Type", "application/x-www-form-urlencoded")
+		req.Header.Add("X-Sumo-Fields", fields)
+	} else if pipeline == MetricsPipeline {
+		// ToDo: Implement metrics pipeline
```

Comment:
Why not return an error saying not implemented?

Reply:
done